dotCMS/src/main/java/com/dotcms/vanityurl/filters/VanityUrlRequestWrapper.java

```diff
@@ -36,12 +35,23 @@ public VanityUrlRequestWrapper(final HttpServletRequest request, final VanityUrl
         super(request);
 
         final boolean vanityHasQueryString = UtilMethods.isSet(vanityUrlResult.getQueryString());
-        
-        this.newQueryString = vanityHasQueryString && UtilMethods.isSet(request.getQueryString())
-                        ? request.getQueryString() + "&" + vanityUrlResult.getQueryString()
-                        : vanityHasQueryString 
-                            ? vanityUrlResult.getQueryString()
-                            : request.getQueryString();
+
+        final StringBuilder params = new StringBuilder();
+        params.append(request.getQueryString());
+        final Map<String, String> vanityParams = convertURLParamsStringToMap(vanityUrlResult.getQueryString());
+        final Map<String, String> requestParams = convertURLParamsStringToMap(request.getQueryString());
+        if(vanityHasQueryString){
+            for (final Map.Entry<String,String> entry : vanityParams.entrySet()) {
+                final String key = entry.getKey();
+                final String value = entry.getValue();
+                //add to the request.getQueryString() the vanity parameters that are not already present, the key and value must not be the same
+                if(!requestParams.containsKey(key) || !requestParams.get(key).equals(value)){
+                    params.append("&" + key + "=" + value);
```

Comment:
Maybe we should check first if `request.getQueryString()` is empty, and in that case the first key/value pair added to the `params` string shouldn't include the "&" character?